### PR TITLE
Add check for tag map

### PR DESCRIPTION
**Pull request by jamie-wearsafe**
Updated 'enabled' flag to check for both 'true' and "true.
Updated the data source to only create a list of tags as maps if an additional tag map is supplied.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -88,7 +88,7 @@ locals {
 }
 
 data "null_data_source" "tags_as_list_of_maps" {
-  count = "${local.enabled ? length(keys(local.tags)) : 0}"
+  count = "${length(keys(var.additional_tag_map)) > 0 && local.enabled == true ? length(keys(local.tags)) : 0}"
```
Comment (aknysh) [inline main.tf]: @jamie-wearsafe  thanks for the PR.
Output `tags_as_list_of_maps` is not related to the input variable `additional_tag_map`.
Output `tags_as_list_of_maps` is used to convert ALL tags into a different format (list of maps) that are required by some AWS resources (not all) regardless whether the input variable `additional_tag_map` provided or not (empty).
The input variable `additional_tag_map` is used to add additional tags to all resources in the format {key: xxxx, value: yyyy}.

`data "null_data_source" "tags_as_list_of_maps"` just converts ALL tags (generated and provided) to the same format {key: xxxx, value: yyyy} AND appends additional tags from `additional_tag_map` if provided.

**Comment by maximmi** [comment]
@jamie-wearsafe is it still actual?

**Comment by jamie-wearsafe** [comment]
> @jamie-wearsafe is it still actual?

Is, which thing still actual :)

**Comment by maximmi** [comment]
@jamie-wearsafe are you able to address Andrew's comments?

**Pull request closed without merging**